src/main/java/com/faforever/client/leaderboard/LeaderboardController.java

```diff
@@ -0,0 +1,369 @@
+    xAxis.setCategories(categories);
+
+    subdivisions.stream().filter(subdivision -> subdivision.getDivision().getIndex() == 1).forEach(firstTierSubDivision -> {
+      XYChart.Series<String, Integer> series = new XYChart.Series<>();
+      series.setName(firstTierSubDivision.getNameKey());
+      subdivisions.stream()
+          .filter(subdivision -> subdivision.getIndex() == firstTierSubDivision.getIndex())
+          .forEach(subdivision -> leaderboardService.getSizeOfDivision(subdivision).thenAccept(size -> {
+            XYChart.Data<String, Integer> data = new XYChart.Data<>(i18n.get(subdivision.getDivisionI18nKey()), size);
+            Text label = new Text();
+            label.setText(subdivision.getNameKey());
+            label.setFill(Color.WHITE);
+            data.nodeProperty().addListener((observable, oldValue, newValue) -> {
+              if (leagueEntry != null && subdivision.equals(leagueEntry.getSubdivision())) {
+                newValue.pseudoClassStateChanged(NOTIFICATION_HIGHLIGHTED_PSEUDO_CLASS, true);
+              }
+              addNodeOnTopOfBar(data, label);
+            });
+            JavaFxUtil.runLater(() -> series.getData().add(data));
+          }));
+      JavaFxUtil.runLater(() -> ratingDistributionChart.getData().add(series));
+    });
+    leaderboardService.getTotalPlayers(season.getId()).thenAccept(totalPlayers ->
+        JavaFxUtil.runLater(() -> xAxis.labelProperty().setValue(i18n.get("leaderboard.totalPlayers", totalPlayers))));
+  }
+
+  private void addNodeOnTopOfBar(XYChart.Data<String, Integer> data, Node nodeToAdd) {
+    final Node node = data.getNode();
+    node.parentProperty().addListener((ov, oldParent, parent) -> {
+      if (parent == null) {
+        return;
+      }
+      Group parentGroup = (Group) parent;
+      ObservableList<Node> children = parentGroup.getChildren();
+      if (!children.contains(nodeToAdd)) {
+        children.add(nodeToAdd);
+        nodeToAdd.setViewOrder(-0.5);
+      }
+    });
+
+    JavaFxUtil.addListener(node.boundsInParentProperty(), (ov, oldBounds, bounds) -> {
+      nodeToAdd.setLayoutX(Math.round(bounds.getMinX() + bounds.getWidth() / 2 - nodeToAdd.prefWidth(-1) / 2));
+      nodeToAdd.setLayoutY(Math.round(bounds.getMaxY() - nodeToAdd.prefHeight(-1) * 0.5));
+    });
+  }
+
+  @NotNull
+  private StringConverter<SubdivisionBean> divisionStringConverter() {
+    return new StringConverter<>() {
+      @Override
+      public String toString(SubdivisionBean subdivision) {
+        return i18n.get(subdivision.getDivisionI18nKey()).toUpperCase();
+      }
+
+      @Override
+      public SubdivisionBean fromString(String string) {
+        return null;
+      }
+    };
+  }
+
+  @NotNull
+  private StringConverter<Number> integerStringConverter() {
+    return new StringConverter<>() {
+      @Override
+      public String toString(Number object) {
+        if(object.intValue()!=object.doubleValue())
+          return "";
```

Comment:
Yes, with default behaviour it would frequenty put values like 5.5 on the axis which makes no sense when we have integers

Reply:
Setting tickUnit in the fxml didn't work?

Reply:
I didn't know about tickUnit. However it doesn't seem to respect that setting...

Reply:
I see, I have to disable autoranging. This also means that I manually, but still automatically have to set the upper bound and the tickUnit to sensible levels otherwise this happens:  
![grafik](https://user-images.githubusercontent.com/52536103/143650869-a1fece02-f197-4ba2-8710-b534fe83c09d.png)
Seems like we need some code for the logic anyway. Which way do you prefer? I think I prefer the string formatter more, because the intention is to get only integers on the axis and the string formatter makes that more clear.

Reply:
Yeah then just the string formatter works. Is annoying it is needed like that though

Reply:
It's not the first thing that is annoying about javafx :P